pkg/cmd/options.go

```diff
@@ -43,6 +44,7 @@ func init() {
 	scheme = runtime.NewScheme()
 	utilruntime.Must(apisconfig.AddToScheme(scheme))
 	utilruntime.Must(v1alpha1.AddToScheme(scheme))
+	install.Install(scheme)
```

Comment:
I can't see why you need to add those types to this `scheme` since it is only used to decode the [service config](https://github.com/gardener/gardener-extension-shoot-cert-service/blob/master/pkg/apis/config/v1alpha1/types.go#L27). Can you please double check?

Reply:
Now I understand the problem. If I drop the `install.Install(scheme)`line in the code and
the `apiVersion` line is missing in the shoot spec, the providerConfig is silently ignored.

```yaml
apiVersion: core.gardener.cloud/v1beta1
kind: Shoot
...
spec:
  extensions:
  - providerConfig:
      kind: CertConfig      
      apiVersion: service.cert.extensions.gardener.cloud/v1alpha1
      dnsChallengeOnShoot:
        enabled: true
        namespace: default
```
So this was my configuration error. Generally it would be good to let the validation fail, if a providerConfig has the `kind` and `apiVersion` fields missing.

Reply:
I have dropped the `install.Install(scheme)`line